### ISSUE TITLE
CheckArrayIndexValueCodeRefactoringProvider disabled when typing opening bracket

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#5169]
VS Mac 7.6.0.1796.
Create a C# console project. Type in `args[` as shown below.

```
	class MainClass
	{
		public static void Main(string[] args)
		{
			Console.WriteLine("Hello World!");
			args[
		}
	}
```

See an exception bar about the CheckArrayIndexValueCodeRefactoringProvider being disabled due to an error.

Show stack trace does not seem to show anything. I will open a different bug for this other problem.

IDE log:

```
Roslyn error: Extension_Exception CheckArrayIndexValueCodeRefactoringProvider : Specified argument was out of the range of valid values.
Parameter name: index
  at Microsoft.CodeAnalysis.SeparatedSyntaxList`1[TNode].get_Item (System.Int32 index) [0x0003c] in /_/src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs:103 
  at Microsoft.CodeAnalysis.SeparatedSyntaxList`1[TNode].First () [0x00000] in /_/src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs:182 
  at RefactoringEssentials.CSharp.CodeRefactorings.CheckArrayIndexValueCodeRefactoringProvider+<ComputeRefactoringsAsync>d__0.MoveNext () [0x00288] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/CheckArrayIndexValueCodeRefactoringProvider.cs:44 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
```

> VS bug [#637460](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/637460)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.348 (master)

Author: Mike Krüger
Commit: f140fc09e226773b275230fdc84dc2803d6af370 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.1812 (release-7.6)

Pull Request #5202 merged by: Lluis Sanchez
Author: mono
Commit: 62f260bf1c1862803d20e67052c19baf19d0bf7c (mono/monodevelop)
